lib/pf/UnifiedApi/Controller/Config/System.pm

```
@@ -97,17 +96,30 @@ sub put_dns_servers {
     }
 }
 
+sub _get_hostname {
+    my ($self) = @_;
+    my $hostnamectl = pf_run("hostnamectl");
+    my $static_hostname;
+    for my $line (split(/\n/, $hostnamectl)) {
+	if($line =~ /^\s*Static hostname:\s*([a-zA-Z0-9]+)/) {
+	    $static_hostname = $1;
+	    last;
+	}
+    }
+    return $static_hostname;
```

we should probably raise an error if $static_hostname is empty or null.